test-cypress/cypress/commands/api/recording.ts

```diff
@@ -0,0 +1,132 @@
+// load the global Cypress types
+/// <reference types="cypress" />
+
+import {v1ApiPath, uploadFile, DEFAULT_DATE } from "../server";
+import {logTestDescription} from "../descriptions"; 
+
+let lastUsedTime = DEFAULT_DATE;
+
+Cypress.Commands.add("uploadRecording", (cameraName: string, details: thermalRecordingInfo, log = true) => {
+  const data = makeRecordingDataFromDetails(details);
+
+  logTestDescription(`Upload '${JSON.stringify(details)}' recording to '${cameraName}'`,
+    { camera : cameraName, requestData: data},
+    log
+  );
+
+  const fileName = 'invalid.cptv';
+  const url = v1ApiPath('recordings');
+  const fileType = 'application/cptv';
+
+  uploadFile(url, cameraName, fileName, fileType, data);
+  // must wait until the upload request has completed
+  cy.wait(['@addRecording']);
+});
+
+interface trackData {
+  start_s? : number,
+  end_s? : number,
+  confident_tag?: string,
+  confidence?: number,
+}
+
+interface algorithmMetadata {
+  model_name? : string
+}
+
+interface thermalRecordingMetaData {
+  algorithm? : algorithmMetadata,
+  tracks : trackData[]
+}
+
+interface thermalRecordingData {
+  type: "thermalRaw",
+  recordingDateTime: string, 
```

Comment:
This is totally an optional style thing, but when I have a string in a particular format, or a number that I know represents an id of a particular type, I like to create a type alias for it.  Eg: `type IsoFormattedDateString = string`, or `type UserId = number` - and then use those aliases in interfaces to provide additional information about what is expected.

Reply:
Like